lib/eventFactory.js

```diff
@@ -772,6 +771,14 @@ class EventFactory extends BaseFactory {
                         }
                     }
 
+                    if (config.baseBranch) {
+                        modelConfig.baseBranch = config.baseBranch;
+                    } else if (prInfo && prInfo.baseBranch) {
+                        modelConfig.baseBranch = prInfo.baseBranch;
+                    } else if (p.scmRepo && p.scmRepo.branch) {
+                        modelConfig.baseBranch = p.scmRepo.branch;
+                    }
```

Comment:
When triggered by a webhook, or when there is a parentEvent such as a restart case, the config value is set in the first scenario. When a PR event is begun from the UI, the PR's target is set in the second scenario. For regular events initiated through the UI, the pipeline's branch is set in the third scenario.

Reply:
perhaps leave this as a comment around the code block for documentation ?

Reply:
OK! I added the comment.